### ISSUE TITLE
Idea for first script?

### ISSUE THREAD

**Issue by FlxVctr** [OII-SDP-2016/oii_how_to_learn_to_code#13]
I'd like something that introduces a module for some API or webscraping (beautiful soup or tweepy for example) to be used to solve some simple task in the first script before people go on to learn pandas. Any ideas?

**Comment by cuducos** [comment]
I like [Tweepy's Getting Started](http://docs.tweepy.org/en/v3.5.0/getting_started.html), but it might be too simple (although it requires users to register an app with Twitter and stuff, and then people might prefer to start [here](http://pythoncentral.io/introduction-to-tweepy-twitter-for-python/)).

Long time ago I did this [Scrapy tutorial](https://youtu.be/758KrjCgkN8) and it was interesting. But honestly it might be very outdated…

**Comment by FlxVctr** [comment]
Scrapy looks super interesting, so we should include it in the extras, but I'm not familiar with it. Maybe let's do the Tweepy's Getting Started and give some more complex task to solve with help of the Tweepy documentation?

**Comment by cuducos** [comment]
Sounds good to me! Scrapy to the _Extras_, Tweepy plus suggested tasks to the main _road map_ ; )

**Comment by FlxVctr** [comment]
Hm, but still don't have an idea for the "more complex task". Any suggestions?

**Comment by cuducos** [comment]
I had in mond something using the stream, but not sure what… maybe choose a hashtag and download some tweets with that hashtag… does that work?

**Comment by FlxVctr** [comment]
Yes, that works. I'm just wondering how to persist them. Or just print to screen? Kind of "Your self-written Twitter-Client"?

**Comment by FlxVctr** [comment]
As a bonus: A script that let's them tweet that they are doing the course at the moment :)

**Comment by FlxVctr** [comment]
Only problem: It's kind of messy to set up the credentials ... and you need a Twitter account :/

**Comment by cuducos** [comment]
>  I'm just wondering how to persist them.

I'd go for just loading data and printing stuff on the screen. But maybe a simple database with [peewee](https://github.com/coleifer/peewee) and SQLite could be useful— I'm just not sure about how simple it actually will be for beginners.

> As a bonus: A script that let's them tweet that they are doing the course at the moment :)

❤️

> It's kind of messy to set up the credentials ... and you need a Twitter account :/

I can write a tutorial on that, or find one to suggest — what do yo think?

**Comment by FlxVctr** [comment]
Aren't there instructions in the Tweepy docs?

**Comment by FlxVctr** [comment]
Against a database approach. Too complicated for the moment.

**Comment by cuducos** [comment]
> Aren't there instructions in the Tweepy docs?

No, they just say _get a request token from twitter_, but [Twitter has](https://dev.twitter.com/oauth/overview/application-owner-access-tokens).

> Against a database approach. Too complicated for the moment.

👍
